### PR TITLE
User Interface updates

### LINKED ISSUE #62: Minimum master password strength
Master passwords should be of a certain strength, without exception. This is more to stop stupidly simple passwords rather than enforcing crazy-complex ones.

### LINKED ISSUE #519: Disable auto loading of icons
Greetings,
For reasons of my own, I'm looking to replace KeePass. I was recommended Buttercup and after some testing with junk data I decided a proper test would be to import my YEARS worth of history from KeePass and give Buttercup a try for a while as my primary.

I instantly regretted it. To my horror, icons started to appear associated with URL domains. Which means my password application IS TALKING TO THE INTERNET WITHOUT MY EXPLICIT PERMISSION! WTF else is it talking to on the network that it isn't telling me about???

Even WORSE I have sites that I no longer associate with, but still have accounts listed in my KeePass. Please please PLEASE tell me that the icons came from some general icon website and that Buttercup did not just query the sites directly. Please tell me that this app didn't just query websites that I no longer wish to be associated with from my home IP WITHOUT MY EXPRESS PERMISSION!

A password vault is supposed to keep my information safe. Not blab freely on the internet! 
How do I disable the auto-loading of icons (and/or disabling of any other kind of internet access without my explicit permission)?
Thanks.

### LINKED ISSUE #531: Password strength meter for new archive password
When changing an archive's password (or when creating it for the first time), show a strength meter.

### PR DESCRIPTION
**Pull request by sallar**
- Remove Sass
- Update dependencies
- Remove flow
- Redesign Password dialog
- Improve icon security

Fixes #519
Fixes #62 
Fixes #531

### CODE REVIEW

File: src/renderer/components/tree-view/index.js [hunk 1]
```diff
@@ -208,27 +208,30 @@ class TreeView extends PureComponent {
...
-        <Tree
-          draggable
-          showLine={false}
-          expandedKeys={this.props.expandedKeys}
-          selectedKeys={this.props.selectedKeys}
-          autoExpandParent={false}
-          onSelect={this.handleSelect}
-          onExpand={this.handleExpand}
-          onDrop={this.handleDrop}
+      <React.Fragment>
```
Comment (perry-mitchell) [inline src/renderer/components/tree-view/index.js]: Destructuring?

File: src/renderer/components/update/index.js [hunk 2]
```diff
@@ -123,51 +123,54 @@ class Update extends PureComponent {
...
-            >
-              <Translate i18nKey="update.not-now" />
-            </Button>
-            <Button
-              primary
-              onClick={this.handleDownload}
-              loading={this.state.installing}
-              disabled={this.state.installing}
-            >
+      <React.Fragment>
```
Comment (perry-mitchell) [inline src/renderer/components/update/index.js]: Destructuring?

File: src/renderer/components/workspace.js [hunk 3]
```diff
@@ -1,78 +1,160 @@
...
+      condencedSidebar,
+      archivesLoading,
+      savingArchive,
+      isArchiveSearchVisible,
+      onValidate
+    } = this.props;
+    const { modalRequest } = this.state;
+
+    return (
+      <React.Fragment>
```
Comment (perry-mitchell) [inline src/renderer/components/workspace.js]: Destructuring?

File: src/renderer/components/file-manager/index.js [hunk 4]
```diff
@@ -94,50 +95,61 @@ class FileManager extends PureComponent {
...
-                  disabled={this.state.selectedConfig === null}
-                  onClick={this.handleOpenClick}
-                >
-                  <Translate
-                    i18nKey="cloud-source.open-in-buttercup"
-                    parent="span"
-                  />
-                </Button>
-              </ButtonRow>
+      <React.Fragment>
```
[the remark this answers is not in the document]
Comment (sallar) [reply src/renderer/components/file-manager/index.js]: @perry-mitchell LAZINESS 😂

**Comment by sallar** [comment]
@perry-mitchell done mate. Now using the Fragment shorthand to void importing it (and react has official support for shorthand)

**Pull request merged**